lib/logstash/dead_letter_post_office.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+class LogStash::DeadLetterPostOffice
+
+  def self.logger=(logger)
+    @logger = logger
+  end
+
+  def self.logger
+    @logger ||= Cabin::Channel.get(LogStash)
+  end
+
+  def self.destination=(destination)
+    logger.info("Dead letter events will be sent to \"#{destination.location}\".")
+    @destination = destination
+  end
+
+  def self.<<(events)
+    events = [events] unless events.is_a?(Array)
+
+    events.each do |event|
+      logger.warn("dead letter received!", :event => event.to_hash)
+      event.tag("_dead_letter")
+      event.cancel
+      @destination << event
```

Comment:
What's the thread-safety here? Multiple plugins could be doing this simultaneously. It's possible `@file.puts` is not threadsafe (further down in your File destination), so we might want to protect this with some kind of lock.